content/configuration/externals.md

```diff
@@ -25,18 +29,62 @@ contributors:
 
 **webpack.config.js**
 
-```js
+```javascript
 externals: {
   jquery: 'jQuery'
 }
 ```
 
 This leaves any dependant modules unchanged, i.e. the code shown below will still work:
 
-```js
+```javascript
 import $ from 'jquery';
 
 $('.my-element').animate(...);
 ```
 
-?> TODO: Add more details on the various ways of doing this using an `object`, `function`, etc. Also add how this is connected to [`output.libraryTarget`](/configuration/output#output-librarytarget) and maybe how it's useful when building libraries if this isn't already evident.
+T> __consumer__ here is any end user application that includes the library that you have bundled using webpack.
+
+## Usage
+
+Your bundle which has external dependencies can be used in various module contexts mainly [CommonJS, AMD, global and ES2015 modules](/concepts/modules). The external library may be available in any of the above form but under different variables.
+
+
+`externals` supports the following module contexts
+
+  * __global__ - An external library can be available as a global variable. The consumer can achieve this by including the external library in a script tag. This is the default setting for externals.
+  * __commonjs__ -  The consumer application may be using a CommonJS module system and hence the external library should be available as a CommonJS module.
+  * __amd__ - Similar to the above line but using AMD module system.
+
+```javascript
+externals : {
```

Comment:
Instead of showing one big example here, I think it would be better to show a specific example under every type (string, array, ...).